src/main/java/org/opentripplanner/astar/spi/AStarEdge.java

```diff
@@ -0,0 +1,13 @@
+package org.opentripplanner.astar.spi;
+
+public interface AStarEdge<
```

Comment:
This need JavaDoc, especially the generic types. I am still not convinced that `Edge` and `Vertex` should be generic types here. This forces the implementation to have a common base-class for these witch is not "reely" necessary. I do not suggest we should remove it in this PR, but we can see how it plays, and change it later if we want to.

Reply:
I removed them, and replaced them with wildcard types, but after that javac crashed with infinite recursion in the type resolver.